src/native/s3select/s3select_napi.cpp

```diff
@@ -258,7 +258,8 @@ S3SelectNapi::S3SelectNapi(const Napi::CallbackInfo& info)
         s3selectEngine::csv_object::csv_defintions csv_defs;
         csv_defs.row_delimiter = GetStringWithDefault(input_serialization_format, "RecordDelimiter", "\n").c_str()[0];
         csv_defs.column_delimiter = GetStringWithDefault(input_serialization_format, "FieldDelimiter", ",").c_str()[0];
-        csv_defs.use_header_info = (0 == GetStringWithDefault(input_serialization_format, "FileHeaderInfo", "IGNORE").compare("USE"));
+        csv_defs.ignore_header_info = (0 == GetStringWithDefault(input_serialization_format, "FileHeaderInfo", "").compare("IGNORE"));
+        csv_defs.use_header_info = (0 == GetStringWithDefault(input_serialization_format, "FileHeaderInfo", "").compare("USE"));
```

Comment:
Where do you use ignore_header_info and use_header_info?
I think the code would be easier to read if you would create a method (like you already created `GetStringWithDefault` that contains these lines and returns a boolean.

Reply:
I don't use them, just pass them to s3select lib in the csv_defs object.
(Because this is the s3select API merging them is not practical)

Reply:
Thanks.
I think you missed my second comment:

> I think the code would be easier to read if you would create a method (like you already created `GetStringWithDefault` that contains these lines and returns a boolean.

Reply:
s3select's csv_defs API has two booleans (ignore_header_info , use_header_info ).
I don't have a way to combine them. There has to be at least two lines, one for each.

Reply:
I thought we can only pass the word we want to compare and the rest will be inside a function (I think it is easier to understand).